### DISCUSSION TITLE
Track performance of CPython on webassembly.

### DISCUSSION THREAD

**Discussion by markshannon** [faster-cpython/ideas#429] [Ideas]
C applications compiled for the browser typically show a slowdown of ~20% or less.
CPython is about 4x slower in the browser.
There are various hypotheses as to why this is, but before we can do anything about it we need numbers and maybe profiling information.
Maybe the pyodide or pyscript teams have more info?

@mdboom 
@antocuni
@hoodmane

**Comment by antocuni** [comment]
Personally, I am aware of the problem but I haven't investigate it yet

**Comment by mdboom** [comment]
Back when I worked on it, the big driver was C function calls where the number of arguments didn't exactly match.  [Hood worked on a fix for that](https://blog.pyodide.org/posts/function-pointer-cast-handling/), and I'm not sure what the next top source of performance problems is at the moment.
